test/unittest.js

```diff
@@ -1,105 +1,185 @@
+// Ensure the stack trace lines numbers are correct on errors
+require('source-map-support').install();
+
 // For the convenience of unit testing, add these to the global namespace
 global._      = require('underscore');
 global.expect = require('chai').expect;
 
 // Unit testing is done against the debug version of the library as it has
 // additional conformance checks that are optimized out of the production
-// library.
-global.Tracer = require('../debug');
-
-var NoopTracerImp;
+// library.  Again, globals are used purely for convenience.
+global.opentracing = require('../debug.js');
+global.Tracer = opentracing.globalTracer();
 
 describe('OpenTracing API', function() {
 
-    // Test that the API layer, while in debug mode, catches API misuse
-    // before the implementation is even invoked (i.e. even with the no-op
-    // implementation).
+    describe('Standalone functions', function() {
+        it('should have a function initGlobalTracer', function() {
+            expect(opentracing.initGlobalTracer).to.be.a('function');
+        });
+        it('should have a function globalTracer', function() {
+            expect(opentracing.globalTracer).to.be.a('function');
+        });
+    });
+    describe('Constants', function() {
+        var constStrings = [
+            'FORMAT_TEXT_MAP',
+            'FORMAT_BINARY',
+            'FORMAT_HTTP_HEADERS',
+            'REFERENCE_CHILD_OF',
+            'REFERENCE_FOLLOWS_FROM',
+        ];
+        _.each(constStrings, function(name) {
+            it(name + ' should be a constant string', function() {
+                expect(opentracing[name]).to.be.a('string');
+            });
+        });
+    });
 
-    describe('surface area', function() {
-        it('should have the required functions on the singleton', function() {
-            expect(Tracer.initGlobalTracer).to.be.a('function');
-            expect(Tracer.initNewTracer).to.be.a('function');
+    describe('Tracer', function() {
+        it('should be a class', function() {
+            expect(opentracing.Tracer).to.be.a('function');
+            expect(new opentracing.Tracer()).to.be.an('object');
         });
 
-        it('should have the required constants', function() {
-            expect(Tracer.FORMAT_TEXT_MAP).to.be.a('string');
-            expect(Tracer.FORMAT_BINARY).to.be.a('string');
-            expect(Tracer.REFERENCE_CHILD_OF).to.be.a('string');
-            expect(Tracer.REFERENCE_FOLLOWS_FROM).to.be.a('string');
+        var tracer = new opentracing.Tracer();
+        var funcs = [
```

Comment:
Is it possible to refactor this file to be reusable for testing API compatibility of other implementations? Similar to this: https://github.com/opentracing/opentracing-python/blob/master/opentracing/harness/api_check.py